pydis_site/apps/content/utils.py

```diff
@@ -0,0 +1,59 @@
+from pathlib import Path
+from typing import Dict, Union
+
+import yaml
+from django.http import Http404
+from markdown2 import markdown
+
+
+def get_category(path: Path) -> Dict[str, str]:
+    """Load category information by name from _info.yml."""
+    if not path.exists() or not path.is_dir():
+        raise Http404("Category not found.")
+
+    return yaml.safe_load(path.joinpath("_info.yml").read_text())
+
+
+def get_categories(path: Path) -> Dict[str, Dict]:
+    """Get all categories information."""
+    categories = {}
+
+    for name in path.iterdir():
+        if name.is_dir():
+            categories[name.name] = get_category(path.joinpath(name.name))
```

Comment:
Isn't `name` already equivalent to `path.joinpath(name.name)`? Also, `name` is a poor choice of a name for a path variable.

Reply:
I agree.